src/pandantic/validators/baseclass.py

```diff
@@ -0,0 +1,8 @@
+from abc import ABC, abstractmethod
+from typing import Any
+
+
+class BaseValidator(ABC):
+    @abstractmethod
+    def validate(self, dataframe: Any) -> Any:
```

Comment:
What do you think about having the function return a bool? I feel semantically that makes the most sense ("validate" -> is it valid -> yes/no -> bool). Also I think without that an Any-> Any base class is somewhat meaningless.

We can more meaningfully build upon the BaseValidator subless (i.e., `pandas` plugin, other features, etc.) with a well defines validation behavior.

Reply:
That said I also see why you may want to support the previous behavior. I personally am in favor of validate and filter being separate functions as well.

Reply:
This does not need to be addressed in this PR, more food for though. Feel free to resolve.

Reply:
I am in favor of changing this in another PR, as the scope for this PR is more or less about the dependency injection pattern and want to avoid the PR becoming very broad scoped.

Reply:
I do agree though, might want to create a ticket for this increment.